File: Ese-Rental-Service-Backend/ese-rental-backend/src/main/java/com/bikalp/eserentalservice/service/UserService.java
package com.bikalp.eserentalservice.service;

import com.bikalp.eserentalservice.dto.UserDto;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;

/**
 * Service interface for User-related operations.
 */
public interface UserService extends UserDetailsService {
    
    /**
     * Create a new user
     * @param userDto the user data
     * @return the created user DTO
     */
    UserDto createUser(UserDto userDto);

    /**
     * Get a user by their ID
     * @param id the user ID
     * @return the user DTO
     */
    UserDto getUserById(Long id);

    /**
     * Get a user by their username
     * @param username the username
     * @return the user DTO
     */
    UserDto getUserByUsername(String username);

    /**
     * Get a user by their email
     * @param email the email
     * @return the user DTO
     */
    UserDto getUserByEmail(String email);

    /**
     * Get all users
     * @return list of user DTOs
     */
    List<UserDto> getAllUsers();

    /**
     * Update a user
     * @param id the user ID
     * @param userDto the updated user data
     * @return the updated user DTO
     */
    UserDto updateUser(Long id, UserDto userDto);

    /**
     * Delete a user
     * @param id the user ID
     */
    void deleteUser(Long id);

    /**
     * Check if a username exists
     * @param username the username to check
     * @return true if the username exists, false otherwise
     */
    boolean existsByUsername(String username);

    /**
     * Check if an email exists
     * @param email the email to check
     * @return true if the email exists, false otherwise
     */
    boolean existsByEmail(String email);
} 

File: Ese-Rental-Service-Frontend/src/api/userService.js
import { axiosInstance } from './axiosInstance';

export const userService = {
    getAllUsers: () => axiosInstance.get('/users'),
    getUserById: (id) => axiosInstance.get(`/users/${id}`),
    createUser: (userData) => axiosInstance.post('/users', userData),
    updateUser: (id, userData) => axiosInstance.put(`/users/${id}`, userData),
    deleteUser: (id) => axiosInstance.delete(`/users/${id}`),
    toggleUserStatus: (id) => axiosInstance.post(`/users/toggle-status/${id}`),
    // Add more user-related API calls as needed
};
